src/vector/_methods.py

```diff
@@ -4405,6 +4362,6 @@ def _flavor_of(*objects: VectorProtocol) -> type[VectorProtocol]:
 
     assert handler is not None
     if is_momentum:
-        return type(handler)
+        return handler.MomentumClass
```

Comment:
I tried doing this without creating `MomentumClass`, but that introduced a lot of edge cases and if conditions in the `_flavor_of` function. Using `MomentumClass` makes it clean and uniform for all the backends.